framework_tvb/tvb/core/entities/storage/burst_dao.py

```diff
@@ -81,8 +81,7 @@ def count_bursts_with_name(self, burst_name, project_id):
             count = self.session.query(BurstConfiguration
                                        ).filter_by(project_id=project_id
                                        ).filter(BurstConfiguration.name.like(burst_name + '%')
-                                                ).filter(not_(BurstConfiguration.name.like(burst_name + '/_%/_%', escape='/'))
-                                       ).count()
+                                                ).count()
```

Comment:
We could use probably `BurstConfiguration.name.notlike`, but you also might be right to remove it.
Have you tried to do a branch from a branch?
A branch from a copy ?
Branch of a branch or a copy?

Reply:
I will try that today.

Reply:
Okay, I checked this now. It does work with notlike in the sense that it doesn't give errors, but that would mean we are getting the number of bursts which don't have a certain name, when in actuality we woud need the number of bursts that do have for example "simulation_1" in their name, so for multiple branches, they would be named "simulation_1_branch1", "simulation_1_branch2" and so on.

Reply:
Let us imagine that we have `simlia` as the first BC.
We branch from it, we want automatic name `simlia_branch1` to be generated.
We branch again from the initial BC `simlia`, and we want automatic name `simlia_branch2` to be generated.
Now, we branch from `simlia_branch1` and the name generated should be `simlia_branch1_branch1`.
If we now branch from `simlia` a 3rd time!!, we want the name `simlia_branch3` to be generated. If you cut that `notlike`, instead it will become the wrong `simlia_branch4`, right?

Thus, pls just correct the previous query to the latest SQL syntax, and do not cut logic.

Reply:
It's true that with "like" instead of "notlike" the behaviour is the one you described, but if I use notlike, then it's not good either: I made a simulation simlia, and then I launched two branches from that and both of them had the name "simlia_branch0" and then I launched a branched from "simlia_branch0" and it was named "simlia_branch0_branch1".

Reply:
Then this should be fixed. 
In https://www.thevirtualbrain.org/user/ Default Project it woks as I describe it.

Reply:
with the current code (so without the second condition), it can not work as I described it.
See the `simlia_branch4` vs `simlia_branch3` last part